lib/core/uri_parser.js

```diff
@@ -687,6 +687,8 @@ function parseConnectionString(uri, options, callback) {
     return callback(new MongoParseError('directConnection option requires exactly one host'));
   }
 
+  parsedOptions._directConnection = Boolean(parsedOptions.directConnection);
+
```

Comment:
isn't the fix here to just remove the code below in the `// NOTE` section?

Reply:
My reasoning for not removing that code is this issue was reported in 3.6, I thought we'd want to fix it there not issue a breaking change. 

> Since changing the starting topology can reasonably be considered a backwards-breaking change, existing drivers SHOULD stage implementation according to semantic versioning guidelines. Specifically, support for the directConnection URI option can be added in a minor release. In a subsequent major release, the default starting topology can be changed to Unknown. Drivers MUST document this in a prior minor release.

[server-discovery-and-monitoring.rst](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst)

Reply:
It's a bug reported in 3.6 using an opt-in feature where we broke compatibility with the existing behavior, I don't think anybody is depending on the bug's behavior. We should just fix it in 3.6 now, and maintain the new behavior as-is in `master`